File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer,Integer> mpp = new HashMap<>();
        for(int i=0 ; i<nums.length;i++)
        {
            int left = target - nums[i];
             if(mpp.containsKey(left))
             {
                return new int []{i,mpp.get(left)};
             }

             else
             mpp.put(nums[i],i);
        }
        return new int[]{-1,-1};
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

    vector <pair<int,int>> arr;

    for(int i=0;i<nums.size();i++)
        arr.push_back({nums[i],i});

        sort(arr.begin(),arr.end());

        vector<int> ans;

        int low=0,high=arr.size()-1;

        while(low<high)
        {
            if(arr[low].first+arr[high].first== target)
            {
                ans.push_back(arr[low].second);
                ans.push_back(arr[high].second);
                break;
            }

            else if(arr[low].first+arr[high].first<target)
            low++;

            else
            high--;
        }
     return ans;
    }
};